todo-backend/todo-backend.yaml

```diff
@@ -0,0 +1,43 @@
+# This configuration file can be used to build and deploy the todo-backend
+# quickstart on OpenShift with the Helm Chart for WildFly.
+build:
+  uri: https://github.com/quickstart/quickstart.git
+  ref: todo-backend
+  mode: bootable-jar
+  env:
+    - name: ARTIFACT_DIR
+      value: todo-backend/target
+    - name: MAVEN_ARGS_APPEND
+      value: -am -pl todo-backend
```

Comment:
One more question to this, before we were using `GIT_CONTEXT_DIR` in template here for this so we are changing this by design?

Because there is a difference between these too procedures:
```
cd todo-backend
mvn install ...
```

```
mvn install -am -pl todo-backend
```
* In the former, if `todo-backend` contains any submodules, these would be build correctly.
* For the latter, if `todo-backend` contains any submodules, these would not be build, only the top `todo-backend` would
  Also in this case, the project root pom.xml is processed too.

As `todo-backend` doesn't have any submodules, we are not hitting this here so this is just thinking out loud about general approach.

I guess the latter is more convenient for development as you rebuild quickstart-parent so one less snapshot to deal with...

Reply:
during development, that is the only way to be able to build the image from a branch (as there are no released pushed to Maven repositories).
If I were using context-dir, I would not be able to verify that this Helm Chart works before the quickstarts have been released...

it's already been discussed and right now, it's not possible to build an Quickstart  image from a branch by pointing to the submodule directly.
To make it work, the quickstart MUST not have snapshots dependencies to WildFLy server and BOM.

Reply:
it makes OpenShift development of the images more complex that it should really be...

Reply:
yes, that's the drawback of using maven in general...

We are dealing with this by using maven repo proxy where we pre-upload all needed snapshot versions but I agree, it is not convenient

Reply:
there is no drawback of using maven here, it's our design that causes issues:
* we rely on SNAPSHOTs of WildFly so we can not built the quickstart from the main branch
* we rely on a parent pom to get the quickstart dependencies so that it's not possible to `context-dir` a single quickstart and have it build in isolation